amundsen_application/static/js/ducks/allTags/sagas.ts

```diff
@@ -1,14 +1,18 @@
 import { SagaIterator } from 'redux-saga';
 import { call, put, takeEvery } from 'redux-saga/effects';
 
+import AppConfig from 'config/config';
 import * as API from './api/v0';
 import { getAllTagsFailure, getAllTagsSuccess } from './reducer';
 import { GetAllTags } from './types';
 
 export function* getAllTagsWorker(): SagaIterator {
   try {
-    const tags = yield call(API.getAllTags);
-    yield put(getAllTagsSuccess(tags));
+    const allTags = yield call(API.getAllTags);
```

Comment:
I have a hard time justifying doing all this extra work in shared code for a concept specific to `TagsList`. This action is dispatched by other components like `TagInput` that need to know about `allTags`. 

`getDerivedStateFromProps` is not necessary, yet I think we went one step to deep for the right place to create `curatedTags` and `otherTags`. Unless there's a reason why this doesn't work with the new libraries -- could we maintain `allTags` as the only necessary piece of state and create `curatedTags` and `otherTags` in `mapStateToProps` for `TagsList`?